tools/pipeline-witness/Azure.Sdk.Tools.PipelineWitness/RunProcessor.cs

```diff
@@ -161,5 +162,74 @@ private async Task<CosmosContainer> GetItemContainerAsync(string containerName)
             var container = cosmosClient.GetContainer(database.Id, containerName);
             return container;
         }
+
+        private async Task UploadLogsBlobAsync(Build build, Timeline timeline)
+        {
+            if (timeline.Records == null)
+            {
+                return;
+            }
+
+            const string timeFormat = @"yyyy-MM-dd\THH:mm:ss.fffffff\Z";
+
+            var hasErrors = false;
+
+            await using var messagesWriter = new StringWriter();
+            foreach(var record in timeline.Records)
+            {
+                var logLines = await this.logProvider.GetTimelineRecordLogsAsync(build, record);
+
+                var lastTimeStamp = record.StartTime;
+
+                for(var lineNumber = 1; lineNumber <= logLines.Count; lineNumber++)
+                {
+                    var line = logLines[lineNumber - 1];
+                    var match = Regex.Match(line, @"^([^Z]{20,28}Z) (.*)$");
+                    var timestamp = match.Success
+                        ? DateTime.ParseExact(match.Groups[1].Value, timeFormat, null, System.Globalization.DateTimeStyles.AssumeUniversal).ToUniversalTime()
+                        : lastTimeStamp;
+                        
+                    var message = match.Success ? match.Groups[2].Value : line;
+
+                    if (timestamp == null)
+                    {
+                        messagesWriter.WriteLine("ERROR: NO LEADING TIMESTAMP");
+                        hasErrors = true;
+                    }
+
+                    await messagesWriter.WriteLineAsync(JsonConvert.SerializeObject(new
+                    {
+                        OrganizationName = "azure-sdk", 
+                        ProjectId = build.Project.Id, 
+                        BuildId = build.Id, 
+                        BuildDefinitionId = build.Definition.Id, 
+                        LogId = record.Log.Id, 
+                        LineNumber = lineNumber, 
+                        Length = message.Length,
+                        Timestamp = timestamp?.ToString(timeFormat),
+                        Message = message,
+                        EtlIngestDate = DateTime.UtcNow.ToString(timeFormat),
+                    }, jsonSettings));
+                    
+                    lastTimeStamp = timestamp;
+                }
+            }
+
+            var blobPathPrefix = $"{build.QueueTime:yyyy/MM/dd}/{build.Project.Name}-{build.Id}";
+            var blobClient = containerClient.GetBlobClient($"/success/{blobPathPrefix}.jsonl");
+
+            if (hasErrors)
+            {
+                var attempt = 1;
+
+                do
+                {
+                    blobClient = containerClient.GetBlobClient($"/error/{blobPathPrefix}-{attempt++}.jsonl");
```

Comment:
Do we expect multiple errors for the same build? Or should this just be either success or error?